pkgs/dart_services/lib/src/common_server.dart

```diff
@@ -202,6 +202,36 @@ class CommonServerApi {
     return ok(version().toJson());
   }
 
+  @Route.post('$apiPrefix/openInIDX')
+  Future<Response> openInIdx(Request request, String apiVersion) async {
+    final code = api.OpenInIdxRequest.fromJson(await request.readAsJson()).code;
+    final idxUrl = Uri.parse('https://idx.google.com/run.api');
+
+    final data = {
+      'project[files][lib/main.dart]': code,
```

Comment:
I assume this is meant to be a string - `'project[files][lib/main.dart]'` - and not a nested set of maps?

Reply:
Yes, this is the correct format.

The IDX API we're using is meant to be used with an HTML form, like this:

```
<form method="post" action="https://idx.google.com/run.api" target="_self">
    <label for="mainpy">main.py:</label>
    <textarea id="mainpy" name="project[files][main.py]"></textarea>

    <label for="devnix">dev.nix:</label>
    <textarea id="devnix" name="project[files][.idx/dev.nix]"></textarea>

    <button type="submit">Open in IDX</button>
</form>
```

Reply:
OK, gotcha.